arangod/Network/ConnectionPool.cpp

```diff
@@ -295,17 +295,22 @@ ConnectionPtr ConnectionPool::selectConnection(std::string const& endpoint,
   auto c = std::make_shared<Context>(createConnection(builder),
                                      now, 1 /* leases*/);
   bucket.list.push_back(c);
+
+  guard.unlock();
+  // continue without the bucket lock
+
+  isFromPool = false;
```

Comment:
How we use this bool&? Is change it without mutex ok?

Reply:
Yes, it is used as a return value